Xamarin.PropertyEditing.Mac/Controls/Custom/CharTextField.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using AppKit;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class CharTextField : NSTextField
+	{
+		string cachedValueString;
+		NSText CachedCurrentEditor { get; set; }
+
+		public event EventHandler ValidatedEditingEnded;
+
+		public override CoreGraphics.CGSize IntrinsicContentSize => new CoreGraphics.CGSize (30, 20);
+
+		public CharTextField ()
+		{
+			BackgroundColor = NSColor.Clear;
+			ControlSize = NSControlSize.Small;
+			StringValue = string.Empty;
+			TranslatesAutoresizingMaskIntoConstraints = false;
+		}
+
+		public override bool ShouldBeginEditing (NSText textObject)
+		{
+			CachedCurrentEditor = textObject;
+			cachedValueString = textObject.Value;
+			CachedCurrentEditor.Delegate = new CharValidationDelegate (this);

```

Comment:
Why do we need to set a new delegate every time we check if we should edit? `ShouldBeginEditing` is only ever going to pass in itself.